lib/model/common/src/errors.ts

```diff
@@ -0,0 +1,135 @@
+/**
+ * Just for convenience, usually it is an Error with name 'AbortError'
+ */
+export class AbortError extends Error {
+  name = 'AbortError';
+}
+
+export function isAbortError(error: unknown): error is Error & { name: 'AbortError' } {
+  return error instanceof Error && error.name === 'AbortError';
+}
+
+export function hasAbortError(error: unknown): boolean {
+  if (!(error instanceof Error)) {
+    return false;
+  }
+
+  return error.name === 'AbortError' || hasAbortError(error.cause);
+}
+
+export function isAggregateError(error: unknown): error is AggregateError {
+  return error instanceof Error && error.name === 'AggregateError';
+}
+
+function stringifyValue(value: unknown): string {
+  if (typeof value === 'string') {
+    return value;
+  }
+
+  if (value && typeof value === 'object') {
+    try {
+      return JSON.stringify(value);
+    } catch (jsonError) {
+      const errorMessage = jsonError instanceof Error ? jsonError.message : String(jsonError);
+      return `Non-serializable object was thrown (JSON.stringify failed: ${errorMessage}): ${String(value)}`;
+    }
+  }
+
+  return String(`Non-Error value (${typeof value}) was thrown: ${value}`);
+}
+
+export function ensureError(value: unknown): Error {
+  if (value instanceof Error) {
+    return value;
+  }
+
```

Comment:
if (isErrorLke(value)) {
const err = new Error(value.message);
err.name = value.name;
err.stack = value.stack;
err.cause = value.cause;
}

Reply:
you can replace this in better place

> And we lost the error type

next string `return new Error(JSON.stringify())`, you also lost type of error type